File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

  const t = Number(input[0])
  const nums = input.slice(1).map(Number)

  const dp =Array.from({length:11},()=>0)
  dp[1] = 1
  dp[2] = 2
  dp[3] = 4

  for(let i=4;i<11;i++){
    dp[i] = dp[i-1] + dp[i-2] + dp[i-3]
  }

  for(let num of nums){
    console.log(dp[num])
  }

File: 백준/Silver/9095. 1， 2， 3 더하기/1， 2， 3 더하기.py
import sys
input = sys.stdin.readline

t = int(input())
num = [int(input()) for i in range(t)]
dp = [0 for i in range(11)]

dp[1] = 1
dp[2] = 2
dp[3] = 4

for i in range(4,11):
  dp[i] = dp[i-1] + dp[i-2] + dp[i-3]

for n in num:
  print(dp[n])